cmd/pdatagen/internal/base_fields.go

```diff
@@ -462,3 +474,77 @@ func (pf *primitiveAsDoubleField) generateCopyToValue(sb *strings.Builder) {
 }
 
 var _ baseField = (*primitiveField)(nil)
+
+type field struct {
+	name            string
+	defaultVal      string
+	testVal         string
+	returnType      string
+	originFieldName string
+	originFullName  string
+	asName          string
+}
+type numberField struct {
+	fields []field
+}
+
+func (pf *numberField) generateAccessors(ms baseStruct, sb *strings.Builder) {
+	for _, field := range pf.fields {
+		sb.WriteString(os.Expand(accessorsNumberTemplate, func(name string) string {
+			switch name {
+			case "structName":
+				return ms.getName()
+			case "fieldName":
+				return field.name
+			case "lowerFieldName":
+				return strings.ToLower(field.name)
+			case "returnType":
+				return field.returnType
+			case "originFieldName":
+				return field.originFieldName
+			case "originFullName":
+				return field.originFullName
+			case "asName":
+				return field.asName
+			default:
+				panic(name)
+			}
+		}))
+		sb.WriteString("\n")
+	}
+}
+
+func (pf *numberField) generateAccessorsTest(ms baseStruct, sb *strings.Builder) {
+	for _, field := range pf.fields {
+		sb.WriteString(os.Expand(accessorsPrimitiveTestTemplate, func(name string) string {
+			switch name {
+			case "structName":
+				return ms.getName()
+			case "defaultVal":
+				return field.defaultVal
+			case "fieldName":
+				return field.name
+			case "testValue":
+				return field.testVal
+			default:
+				panic(name)
+			}
+		}))
+		sb.WriteString("\n")
+	}
+}
+
+func (pf *numberField) generateSetWithTestValue(sb *strings.Builder) {
+	for _, field := range pf.fields {
+		sb.WriteString("\ttv.Set" + field.name + "(" + field.testVal + ")\n")
+	}
+
+}
+
+func (pf *numberField) generateCopyToValue(sb *strings.Builder) {
+	for _, field := range pf.fields {
+		sb.WriteString("\tdest.Set" + field.name + "(ms." + field.name + "())\n")
+	}
```

Comment:
This should be switch on the "type" and call set only for that.

Reply:
Yup, fixed.